helper/plancheck/resource_action.go

```diff
@@ -0,0 +1,18 @@
+package plancheck
+
+// TODO: is this a good name?
+// TODO: document type and all aliases below
+type ResourceActionType byte
```

Comment:
Upstream terraform-json already has the [`Action` and `Actions` types](https://pkg.go.dev/github.com/hashicorp/terraform-json#Action) for some of these -- I wonder if we should follow suit here and rely on those, rather than inventing our own? If we need a simpler "replace" maybe that's something we can submit upstream or should be the only part we are creating.

Personally, I also prefer `string` or otherwise self-describing constant types rather than opaque data. It makes debugging or printing out the constant much friendlier.

Reply:
Right, I added this because of the combinations of the "multi action" actions:
- ResourceActionDestroyBeforeCreate -> `tfjson.ActionDelete + tfjson.ActionCreate`
- ResourceActionCreateBeforeDestroy -> `tfjson.ActionCreate + tfjson.ActionDelete`
- ResourceActionReplace -> `tfjson.ActionDelete + tfjson.ActionCreate` or `tfjson.ActionCreate + tfjson.ActionDelete`

I had it in my list of TODOs to revisit, so I'll work through the options and see if there is a cleaner way to surface this

Reply:
After more discussion and investigation, we opted to leave this enum in and ensure that the documentation explains thoroughly what is being abstracted. Added in c3aefc4 ⚡